docs/development.rst

```diff
@@ -0,0 +1,151 @@
+.. _development-section:
+
+Development documentation
+=========================
+
+High-level architecture overview
+--------------------------------
+
+``Edx-enterprise`` is a pluggable Django application designed to be run within edx-platform. It is shipped with any
+edx-platform install, but can be disabled or even removed from particular Open edX instance.
+
+General direction is to decouple ``edx-enterprise`` from the rest of the edx-platform as much as possible, since
+the overall goal is to eventually make it an independent application or a plugin for a separate admin site, and not be
+part of the edx-platform LMS.
+
+Hence the following must be taken into account:
+
+* When integrating with existing edx-platform features, REST APIs should be preferred over python APIs.
+* In case API in question only exists as python API it should be encapsulated into a module/class on ``edx-enterprise``
+  side, to simplify future migration to REST API.
+* Both REST and python API clients must be covered with extensive unit tests.
+
+Other noteworthy points:
+
+* ``edx-enterprise`` does not interact with Studio.
+* ``edx-enterprise`` supports multiple python and Django versions:
+
+  * Python 2.7 and python 3.5.
+  * Django 1.8, 1.9 and 1.10.
+
+* High test coverage is expected - all new python code must have 100% diff coverage.
+  JS test were only recently introduced, so it's not as strict so far, but aim for 100% diff coverage as well.
+* High code quality standards are expected - both on python and JS sides. Multiple static analyzers are used to enforce
+  it (``pycodestyle``, ``pylint``, ``isort``, ``jshint``, etc.).
+
+Local setup
+-----------
+
+Although ``edx-enterprise`` can't be used outside edx-platform, some development process steps are intended to be
+performed in a standalone virtual environment.
+
+If you have not already done so, create/activate a `virtualenv`_. Unless otherwise stated, assume all terminal code
+below is executed within the virtualenv.
+
+.. _virtualenv: https://virtualenvwrapper.readthedocs.org/en/latest/
+
+Install dependencies
+^^^^^^^^^^^^^^^^^^^^
+
+Dependencies can be installed via the command below.
+
+.. code-block:: bash
+
+    $ make requirements
+
+
+Run tests
+^^^^^^^^^
+
+This application uses `tox`_ to run tests under multiple python and Django version. In addition, the following test
+utilities are used:
+
+* `pytest`_ framework and test runner to run test
+* `pytest-cov`_ plugin for collecting test coverage
+* `diff-cover`_ for analyzing changeset coverage
+
+.. _tox: https://tox.readthedocs.io/en/latest/
+.. _pytest: http://doc.pytest.org/en/latest/
+.. _pytest-cov: https://pypi.python.org/pypi/pytest-cov
+.. _diff-cover: https://github.com/Bachmann1234/diff-cover
+
+`edx-enterprise` Makefile provide multiple shortcuts to running tests, the most noteable are:
```

Comment:
"The `edx-enterprise` Makefile provides multiple shortcuts to running tests. The most notable are:"